examples/arrowir/test_arrowir.cpp

```diff
@@ -0,0 +1,487 @@
+#include "duckdb.hpp"
+
+#include "duckdb/main/client_context.hpp"
+#include "duckdb/planner/operator/list.hpp"
+#include "duckdb/planner/expression/list.hpp"
+#include "duckdb/function/table/table_scan.hpp"
+#include "duckdb/catalog/catalog_entry/table_catalog_entry.hpp"
+
+#include "Schema_generated.h"
+#include "Plan_generated.h"
+#include "Relation_generated.h"
+
+#include "flatbuffers/minireflect.h"
+
+#include "tpch-extension.hpp"
+
+#include <string>
+
+using namespace std;
+
+namespace arrowir = org::apache::arrow::computeir::flatbuf;
+namespace arrow = org::apache::arrow::flatbuf;
+
+static arrow::Type transform_type(duckdb::LogicalType &type) {
+	switch (type.id()) {
+	case duckdb::LogicalTypeId::UTINYINT:
+	case duckdb::LogicalTypeId::USMALLINT:
+	case duckdb::LogicalTypeId::INTEGER:
+	case duckdb::LogicalTypeId::BIGINT:
+		return arrow::Type_Int; // TODO is this correct?
```

Comment:
Unfortunately it doesn't seem like there's a way to represent a specific integer width using the `Type` enum.

It looks like without using the arrow library our options are:

1) make a separate union of compute IR types that mirrors the arrow types except it explicitly enumerates the integer types `Int8`, `UInt8`

2) make a new union in compute IR that contains the structures used to represent different widths, something like:

```
union Type {
  org.apache.arrow.flatbuf.Int,
  org.apache.arrow.flatbuf.FloatingPoint,
  org.apache.arrow.flatbuf.Type,
  org.apache.arrow.flatbuf.Field, // necessary for compound types
}
```

Reply:
@hannesmuehleisen What I said above was wrong actually. The `Type` object is a flatbuffers `union`, and contains the `Int` `table` which has a `bitWidth` field that you should be able to use to differentiate the different integer types. The main problem is that flatbuffers requires `union`s to be embedded in a table to be able to use them.

What this means is that you likely need to return the result of `arrowir::CreateType(...)` from this function instead of the union discriminant value.

Does that make sense?